File: android/app/src/main/java/com/gwangy/nassspandroidsample/SdkHooks.kt
package com.gwangy.nassspandroidsample

object SdkHooks {
    fun describe(format: SampleFormat): String = when (format) {
        SampleFormat.Banner -> banner()
        SampleFormat.Native -> native()
        SampleFormat.Video -> video()
        SampleFormat.RewardVideo -> rewardVideo()
        SampleFormat.InterstitialVideo -> interstitialVideo()
        SampleFormat.HybridWebView -> hybridWebView()
    }

    fun banner(): String = "배너 광고를 붙일 준비가 됨"
    fun native(): String = "네이티브 광고를 붙일 준비가 됨"
    fun video(): String = "동영상 광고를 붙일 준비가 됨"
    fun rewardVideo(): String = "리워드 동영상 광고를 붙일 준비가 됨"
    fun interstitialVideo(): String = "전면 동영상 광고를 붙일 준비가 됨"
    fun hybridWebView(): String = "웹뷰에서 네이티브 브리지를 사용할 준비가 됨"

    fun execute(format: SampleFormat, context: android.content.Context): String = when (format) {
        SampleFormat.Banner -> { NapSspSdkIntegration.bannerView(context); "배너 광고 실행" }
        SampleFormat.Native -> { NapSspSdkIntegration.nativeView(context); "네이티브 광고 실행" }
        SampleFormat.Video -> { NapSspSdkIntegration.videoView(context); "동영상 광고 실행" }
        SampleFormat.RewardVideo -> { NapSspSdkIntegration.rewardVideoView(context); "리워드 동영상 광고 실행" }
        SampleFormat.InterstitialVideo -> { NapSspSdkIntegration.interstitialVideoView(context); "전면 동영상 광고 실행" }
        SampleFormat.HybridWebView -> { "웹뷰 브리지 실행" }
    }

    fun hybridStatus(): String = "NapSsp hybrid bridge ready"
}

File: ios/Sources/NapSspIOSSample/SdkHooks.swift
import Foundation
import UIKit

enum SdkHooks {
    static func describe(_ format: SampleFormat) -> String {
        switch format {
        case .banner: return banner()
        case .native: return native()
        case .video: return video()
        case .rewardVideo: return rewardVideo()
        case .interstitialVideo: return interstitialVideo()
        case .hybridWebView: return hybridWebView()
        }
    }

    static func banner() -> String { "배너 광고 준비됨" }
    static func native() -> String { "네이티브 광고 준비됨" }
    static func video() -> String { "동영상 광고 준비됨" }
    static func rewardVideo() -> String { "리워드 광고 준비됨" }
    static func interstitialVideo() -> String { "전면 광고 준비됨" }
    static func hybridWebView() -> String { "웹뷰 브리지 준비됨" }

    static func execute(_ format: SampleFormat) -> String {
        guard let rootVC = rootViewController() else {
            return "루트 뷰컨트롤러를 찾지 못함"
        }

        switch format {
        case .banner:
            _ = NapSspSdkIntegration.banner(rootVC: rootVC)
            return "배너 광고 실행"
        case .native:
            _ = NapSspSdkIntegration.native(rootVC: rootVC)
            return "네이티브 광고 실행"
        case .video:
            _ = NapSspSdkIntegration.video(rootVC: rootVC)
            return "동영상 광고 실행"
        case .rewardVideo:
            NapSspSdkIntegration.rewardVideo(rootVC: rootVC)
            return "리워드 동영상 광고 실행"
        case .interstitialVideo:
            NapSspSdkIntegration.interstitialVideo(rootVC: rootVC)
            return "전면 동영상 광고 실행"
        case .hybridWebView:
            return "웹뷰 브리지 실행"
        }
    }

    private static func rootViewController() -> UIViewController? {
        UIApplication.shared.connectedScenes
            .compactMap { $0 as? UIWindowScene }
            .flatMap(\.windows)
            .first(where: \.isKeyWindow)?
            .rootViewController
    }

    static func hybridStatus() -> String {
        "NapSsp hybrid bridge ready"
    }
}
